src/unchunker.ts

```diff
@@ -111,16 +198,19 @@ class ChunkCollector {
     }
 
     /**
-     * Merge the messages.
+     * Merge the chunks into a single message.
      *
      * Note: This implementation assumes that no chunk will be larger than the first one!
      * If this is not the case, an error may be thrown.
      *
-     * @return An object containing the message as an `Uint8Array`
-     *         and a (possibly empty) list of context objects.
+     * @return An object containing the message and a (possibly empty) list of context
+     *          objects.
      * @throws Error if message is not yet complete.
      */
-    public merge(): {message: Uint8Array, context: any[]} {
+    public merge(): {message: any, context: any[]} {
+        // TODO: Performance. Merging all at once in the end is usually slow compared
+        //       to merging with each chunk.
```

Comment:
Is there something that still needs to be done / benchmarked here?

Reply:
I'd suggest doing that in a separate PR. But yes, this could be quite a performance improvement. I used the same approach in RAWRTC (merging at the end) and it turned out to be significantly slower (a bunch of seconds when merging 1 GiB at once) than merging with each chunk.